### ISSUE TITLE
FEAT: Folders support

### ISSUE THREAD

**Issue by I2rys** [dbgate/dbgate#274]
**Is your feature request related to a problem? Please describe.**
It's quiet annoying If you have a lot of database to manage mostly If you can't organize & filter them(The color's is not enough too).

**Describe the solution you'd like**
Being able to create folders(Create, Delete, Rename) and drag & drop databases on them, basicly just like a normal file explorer.

**Describe alternatives you've considered**
Being able to use tags and filter tags.

**Comment by GitClickOk** [comment]
You can name it "Tags" or "Groups" too. But "Folders" is great too if we can have subfolders.

**Comment by wallysonn** [comment]
> **Sua solicitação de recursos está relacionada a um problema? Por favor, descreva.** É silencioso irritante Se você tem um monte de banco de dados para gerenciar principalmente Se você não pode organizá-los e filtrar(A cor não é suficiente também).
> 
> **Descreva a solução que você gostaria** de ser capaz de criar pastas (Criar, excluir, renomear) e arrastar e soltar bancos de dados neles, basicamente como um explorador de arquivos normal.
> 
> **Descreva alternativas que você considerou** ser capaz de usar tags e etiquetas de filtro.

Please, this is very important.

**Comment by qlaffont** [comment]
I'm starting to work on it but it seems that svelte transpilation is not working.

```bash
src/tabs/DiagramTab.svelte
27: export let conid;
28: export let database;
29: export let initialArgs;
               ^
30: export const activator = createActivator('DiagramTab', true);
31: $: setEditorData($modelState.value);
[!] (plugin svelte) CssSyntaxError: /home/quentin/Documents/dev/qlaffont/dbgate/packages/web/src/Screen.svelte:1:3: Unknown word
src/Screen.svelte
CssSyntaxError: /home/quentin/Documents/dev/qlaffont/dbgate/packages/web/src/Screen.svelte:1:3: Unknown word
    at Input.error (/home/quentin/Documents/node_modules/postcss/lib/input.js:148:16)
    at Parser.unknownWord (/home/quentin/Documents/node_modules/postcss/lib/parser.js:540:22)
    at Parser.other (/home/quentin/Documents/node_modules/postcss/lib/parser.js:164:12)
    at Parser.parse (/home/quentin/Documents/node_modules/postcss/lib/parser.js:72:16)
    at parse (/home/quentin/Documents/node_modules/postcss/lib/parse.js:11:12)
    at new LazyResult (/home/quentin/Documents/node_modules/postcss/lib/lazy-result.js:133:16)
    at Processor.process (/home/quentin/Documents/node_modules/postcss/lib/processor.js:28:14)
    at transformer (/home/quentin/Documents/dev/qlaffont/dbgate/node_modules/svelte-preprocess/dist/transformers/globalStyle.js:62:72)
    at transform (/home/quentin/Documents/dev/qlaffont/dbgate/node_modules/svelte-preprocess/dist/autoProcess.js:37:12)
    at style (/home/quentin/Documents/dev/qlaffont/dbgate/node_modules/svelte-preprocess/dist/autoProcess.js:191:33)


[2022-11-03 08:17:36] waiting for changes...
```

-> https://github.com/qlaffont/dbgate

**Comment by janproch** [comment]
Hm, I haven't seen similar error yet.

Has it ever worked before for you? And what command do for starting UI you use? yarn start:web ?

**Comment by qlaffont** [comment]
It was the first time I start the project, And yes i have started the server with yarn start:web
node version : v14.18.2

**Comment by janproch** [comment]
I use node version 16.14.2 now, but it shouldn't depend on it

And did you install your dependencies with `yarn install` or `npm install` ?

**Comment by qlaffont** [comment]
I will try to retry with your node-version yarn install

**Comment by qlaffont** [comment]
seems to not work too ...

**Comment by qlaffont** [comment]
Static hostname: pop-os
       Icon name: computer-laptop
         Chassis: laptop
Operating System: Pop!_OS 22.04 LTS          (Based on Ubuntu)      
          Kernel: Linux 6.0.3-76060003-generic
    Architecture: x86-64
 Hardware Vendor: Dell Inc.
  Hardware Model: XPS 15 9500

**Comment by janproch** [comment]
I tried it with clean clone on my Ubuntu now, and it works. But there are lot of warnings from Svelte, maybe some of them could cause errors in some situations, so I will try to eliminate these warnings, I will let you know then.

**Comment by janproch** [comment]
OK, decided to upgrade to newest Svelte toolkit, but I got stuck in this

Anyway, I came across the same problem, that you encountered, and I believe, that it is solved it newest master

**Comment by qlaffont** [comment]
Works thanks, I will try to work on it ^^

**Comment by janproch** [comment]
Folders feature is now released in BETA, thanks @qlaffont

**Comment by qlaffont** [comment]
@janproch as I tested on beta. Parent field is working but d-and works for me. 

As I can see for d-and-d, we should be able to add connection to parent if we drag it on sub child list. 
Because at this time, you can add it only if your drop on parent name.

**Comment by janproch** [comment]
D&D is now fixed.

I have also changed the way, how new folder is created. I added new button for this in connection list.

So I have removed "Parent folder" field in connection details, as it is a bit user unfriendly to creating folders as side effect of editing connections. But I am open to discuss this topic, this modification is in a standalone commit, so it is easy to revrt this modification.

**Comment by qlaffont** [comment]
For me it's the perfect integration 👌

**Comment by janproch** [comment]
Released in 5.1.6, thanks @qlaffont
